.github/workflows/build_and_test.yml

```
@@ -175,10 +215,11 @@ jobs:
         run: cargo deny check --hide-inclusion-graph
 
   build-neon:
-    needs: [ check-permissions, tag ]
+    if: ${{ always() }}
```

Just `always()` maybe a bit too much, we don't want to run these jobs if something has failed or has been cancelled, I think an alternation like this should do the trick

```suggestion
    if: always() && contains(fromJSON('["success", "skipped"]'), needs.build-buildtools-image.result)
```

A side note, I don't really like how it looks, complicates the workflow, and how fragile it is, but this is the only way I know. I hope we can improve in the future